core/Pos/Binary/Class/Core.hs

```diff
@@ -68,6 +70,16 @@ matchSize requestedSize lbl actualSize =
   when (actualSize /= requestedSize) $
     fail (lbl <> " failed the size check. Expected " <> show requestedSize <> ", found " <> show actualSize)
 
+-- | Tries to decode Tag 24 (CBOR Data Item) from a ByteString blob,
```

Comment:
@neongreen Ah, I must have missed that YouTrack issue! 

> That's right, the proper use of the tag is in front of (consider to be enclosing if you think of a term tree) the the bytestring. So not inside.

I'm afraid I'm out of time for today, but for the sake of my own understanding, why the following wouldn't be correct? (which is essentially the current implementation proposed in my PR):

```
toLazyByteString (encodeTag 24 <> encode innerType)
```

Here I am indeed putting the tag in front of the `encode innerType`, which is the `Encoding` of the inner `ByteString`. I totally agree that @neongreen is the most readable implementation and we should probably go with that!